standards/pre-rootfs-environment.rst

```diff
@@ -0,0 +1,201 @@
+Pre-rootfs environment
+======================
+
+Introduction
+------------
+
+Halium-standard Linux distributions have different needs than their "Desktop" or "Laptop" counterparts. Often, a Halium-standard distribution will need to start the Android container before it is able to complete its initialization. The Halium pre-rootfs environment ensures that the distribution will be able to successfully start the Android container and continue booting.
+
+Scope
+-----
+
+This document specifies the minimum requirements for a pre-rootfs environment that is able to boot Halium-standard Linux distributions. By doing so, it also specifies what a maintainer of a Halium-standard distribution may expect when their software begins to run.
+
+Halium-standard distribution maintainers may wish to skip to `Annex A: Summary for HSDs`_ to understand the parts of this standard that apply to them.
+
+Terms and definitions
+---------------------
+
+This document follows the general terms, definitions, and expressions defined in the Halium :ref:`standard-terms-and-conventions` with the following additions:
+
+Pre-rootfs environment
+    Code that executes before the init system of the HSD begins to run.
+
+Rootfs
+------
+
+The pre-rootfs environment shall mount the HSD rootfs and its desired Android system.img from persistent storage. The rootfs can take the following formats:
+
+* ``rootfs.img``, an image file located on the Device's ``/data`` partition. It will be accompanied by ``system.img``, which will be a sibling in its directory.
+* ``halium-rootfs/``, a directory (or symbolic link to a directory) on the device's ``/data`` partition. This directory will contain the ``system.img`` at ``var/lib/lxc/android/system.img``.
```

Comment:
It could be preferred to allow image file and directory names to be received from kernel cmdline (for example osname=ubports) and defaulted to rootfs.img/halium-rootfs. This way makes it easier to keep multiple OSes on single device, only boot image would need to be changed.

Reply:
Does multiboot need to be required? The rootfs format is up to the environment, I'd say that the selection of any format would be as well.